### DISCUSSION TITLE
Config a route that can handle multiple $POSTs with long running calls?

### DISCUSSION THREAD

**Discussion by aktiver** [fastapi/fastapi#8601] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
Ques: how does one make a FastAPI route that can handle multiple $POSTs with long running calls?
```

### Description

Example:
[1,2,3,4,5]   ->  /myRoute

5 posts all at once and each of the processes will run inside the route, in parallel.  

### Operating System

Linux

### Operating System Details

Alpine 

### FastAPI Version

Latest stable 

### Python Version

3.9

### Additional Context

None.

**Comment by rafsaf** [comment, marked as answer]
This is the requirement for almost every single API one can create. What exactly a problem can you see with multiple POSTs at once? You don't need specific implementation of FastAPI to achieve this. It's looks more problem with configuration of your web server (`workers` param in uvicorn/gunicorn, `processes` in nginx unit), by default it is often 1 process.

**Comment by nerdstrike** [comment]
As @rafsaf points out, this problem is as old as web servers.
First you need pre-forks/worker threads courtesy of the web container, i.e. uvicorn/gunicorn.
Then you need to ensure there is no blocking behaviour that might prevent the different threads from executing at the same time. Liberal use of async in your routes allows the host to park the thread while something slow happens (such as writing to disk or a database), thereby increasing the number of requests that you can pick up at once, even though it may not make them happen simultaneously.

Beware that long running intensive requests will occupy a worker for the duration of the request, and it is very easy to get into trouble when receiving lots of requests. The master thread will buffer up requests but if your throughput is too low, it will begin rejecting requests. You may also find that inexensive requests to other routes get denied or become very slow. One practical solution for long duration and high traffic is to set up an offline queuing system behind the scenes and tell the client where to go to get the result when it becomes ready. That way the web server is free to handle more requests, because it has offloaded the slow operation. See https://testdriven.io/blog/fastapi-and-celery/ for an example of using `celery` for background processing.
